protocol/v2/ssv/validator/committee.go

```diff
@@ -178,6 +194,13 @@ func (c *Committee) StartDuty(logger *zap.Logger, duty *spectypes.CommitteeDuty)
 
 	}
 
+	// Stop all old committee runners, when new runner is created
+	go func() {
+		if err := c.StopOldRunners(logger, duty.Slot); err != nil {
+			logger.Error("couldn't stop old committee runners", zap.Uint64("current_slot", uint64(duty.Slot)), zap.Error(err))
+		}
+	}()
```

Comment:
why is a goroutine needed here?

Reply:
I decided to call it in goroutine, because we have `mtx.Lock()` inside`StopOldRunners`. The problem is that the mutex is already locked by `StartDuty` and it's going to be deadlock.
So I see we have 2 options:
- call inside goroutine and it will take the resource after `StartDuty` execution is finished
- call it synchronously, but remove the lock inside of `StopOldRunners`. 

I prefer to keep `mtx.Lock()` usage `StopOldRunners()` to prevent data race.

Let me know what do you think about it